### DISCUSSION TITLE
Running the dungeon generation in a Coroutine

### DISCUSSION THREAD

**Discussion by foobraco** [OndrejNepozitek/Edgar-Unity#118] [Q&A] [answered]
Right now generating a level is a blocking action, but we would like to have some sort of animation while the level is being generated. do you know if there is a way to get the generator running in a coroutine to make it non-blocking?

**Comment by OndrejNepozitek** [comment, marked as answer]
Hey, a non-blocking generator is a PRO feature right now so I don't plan to add it to the free version. If you have the PRO version, you can for example check the `GungeonGameManager` class which has this code inside.

**Reply by foobraco** [reply]
Thanks for the quick response! I'll check out the PRO version!
